app/src/main/java/space/taran/arkmemo/data/repositories/TextNotesRepo.kt

```diff
@@ -0,0 +1,151 @@
+        }
+    }
+
+    suspend fun delete(note: TextNote) = withContext(iODispatcher) {
+        val path = root.resolve("${note.meta?.name}")
+        remove(note)
+        delete(path)
+        propertiesStorage.remove(note.meta?.id!!)
+        propertiesStorage.persist()
+        Log.d("text-repo", "${note.meta?.name!!} has been deleted")
+    }
+
+    suspend fun read() = withContext(Dispatchers.IO) {
+        val notes = mutableListOf<TextNote>()
+        Files.list(root).forEach { path ->
+            if (path.fileName.extension == NOTE_EXT) {
+                try {
+                    val data = StringBuilder()
+                    path.forEachLine {
+                        data.appendLine(it)
+                    }
+                    val size = path.fileSize()
+                    val id = computeId(size, path)
+                    val meta = ResourceMeta(
+                        id,
+                        path.fileName.name,
+                        path.extension,
+                        path.getLastModifiedTime(),
+                        size
+                    )
+                    val titles = propertiesStorage.getProperties(id).titles
+                    val content = TextNote.Content(titles.elementAt(0), data.toString())
+                    val note = TextNote(content, meta)
+                    notes.add(note)
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                }
+            }
+        }
+        Log.d("text-repo", "${notes.size} text note resources found")
+        _textNotes.value = notes
+    }
+
+    private suspend fun write(note: TextNote) = withContext(Dispatchers.IO) {
+        val path = root.resolve("${DUMMY_FILENAME}.${NOTE_EXT}")
+        if (!path.exists()) {
+            try {
+                val lines = note.content.data.split(NEWLINE)
+                path.writeLines(lines)
+                val size = path.fileSize()
+                val id = computeId(size, path)
+                val properties = Properties(setOf(note.content.title), setOf())
+
+                Log.d("text-repo", "initial resource name ${path.name}")
+
+                propertiesStorage.setProperties(id, properties)
+                propertiesStorage.persist()
+
+                val newPath = root.resolve("$id.$NOTE_EXT")
+                if (!newPath.exists()) {
+                    if (path.toFile().renameTo(newPath.toFile())) {
+                        note.meta = ResourceMeta(
+                            id,
+                            newPath.fileName.name,
+                            newPath.extension,
+                            newPath.getLastModifiedTime(),
+                            size
+                        )
+                        Log.d("notes-repo", "resource renamed to ${newPath.name} successfully")
+                    } else delete(path)
+                }
+            } catch (e: Exception) {
+                e.printStackTrace()
+            }
```

Comment:
Can any exceptions really occur here? If `try-catch` is necessary, then better to log at `warning` level, not just print.

@hieuwu Please correct me if I'm wrong.